libs/data-types/README.md

```diff
@@ -1,11 +1,78 @@
-# data-types
+# @coscrad/data-types
+
+## About
 
 This library was generated with [Nx](https://nx.dev).
 
-## Building
+The purpose of this library within our Coscrad monorepo is to define our own
+custom set of data types. These types (and even their validation logic) can be
+shared with the client, and also can be used to generate API documentation. We
+achieve this by providing a set of `Custom Data Type` decorator factory functions,
+along with a utility function `getCoscradDataSchema`, which allows one to retrieve
+the custom data type via Reflection by leveraging
+[reflect-metadata](https://www.npmjs.com/package/reflect-metadata).
+
+Further, we encapsulate validation rules for said data
+types using our own validation library (@coscrad/validation), which includes
+a set of validation decorators and validation functions.
+
+One use of this library is to simplify the management of payload schemas for commands.
+
+## Usage
+
+This library is based on decorators (specifically property decorators) and hence
+it works with classes. To define a schema for a class,
+decorate each property using the appropriate `Custom Data Type` decorator factory
+function. Use the `isOptional` and `isArray` options as needed.
+
+```ts
+class Foo {
+    @NonEmptyString()
+    readonly title: string;
```

Comment:
Do we need to declare if it is public or private?

Ex:

@NonEmptyString()
public readonly title: string;

Reply:
no- public is the default. You only have to declare private.

However, insomuch as a command schema is part of our API, all fields must be public.